### PR TITLE
policy-controller: add pre-delete hook

### PR DESCRIPTION
**Pull request by hectorj2f**
## Description of the change

Add helm hook to pre-delete the leases.

closes https://github.com/sigstore/helm-charts/issues/504

## Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). Where applicable, update and bump the versions in any associated umbrella chart
- [x] Variables are documented in the `values.yaml` and added to the README.md. The [helm-docs](https://github.com/norwoodj/helm-docs) utility can be used to generate the necessary content. Use `helm-docs --dry-run` to preview the content.
- [x] JSON Schema generated.
- [x] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command.

### CODE REVIEW

File: charts/policy-controller/templates/webhook/cleanup-leases.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: leases-cleanup
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "policy-controller.labels" . | nindent 4 }}
+    control-plane: {{ template "policy-controller.fullname" . }}-webhook
+  annotations:
+    "helm.sh/hook": pre-delete
```
[the remark this answers is not in the document]
Comment (hectorj2f) [reply charts/policy-controller/templates/webhook/cleanup-leases.yaml]: 🤔 , I didn't face that race condition but I just changed to `post-delete`.

**Review by vaikas — APPROVED** [review]
Sweet, thanks!

**Comment by eddiezane** [comment]
Talked to Hector on Slack. There are still a few issues here. He's going to label some of the other resources to ensure there are no race conditions on deletion.

**Comment by hectorj2f** [comment]
This is ready for another review.

**Review by priyawadhwa — APPROVED** [review]

**Review by eddiezane — APPROVED** [review]
This looks great and I can no longer reproduce the issue! Thank you!!

**Pull request merged**